File: Difficulty: Easy/Second Largest/second-largest.cpp
class Solution {
  public:
    int getSecondLargest(vector<int> &a) {
        // code here
        
    int n = a.size();
    if (n < 2) return -1;  
     
    // int largest = INT_MIN;
    int fl = INT_MIN;
    int sl = INT_MIN;
    
    for(int i = 0; i < n; i++) {
      if(a[i] > fl) {
        sl = fl;
        fl = a[i];
      } else if(a[i] > sl && a[i] < fl) {
        sl = a[i];
      }
    }
    
    return (sl == INT_MIN) ? -1 : sl; 
    }
};

File: Difficulty: Easy/Second Largest/second-largest.java
class Solution {
    public int getSecondLargest(int[] arr) {
        // code here
        int n = arr.length;
        int first = Integer.MIN_VALUE;
        int sec = Integer.MIN_VALUE;
        
        for(int i = 0; i < n ;i++) {
            if(arr[i] > first) {
                sec = first;
                first = arr[i];
            } else if(arr[i] > sec && arr[i] < first) {
                sec = arr[i];
            }
        }
        if(sec == Integer.MIN_VALUE)
            return -1;
        return sec;
    }
}